tests/forms/test_permission_delete.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""Test permission DeleteForm."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from flask_babel import gettext as _
+
+from xl_auth.permission.forms import DeleteForm
+
+from ..factories import PermissionFactory
+
+
+def test_validate_permission_id_does_not_exist(superuser, permission):
+    """Attempt deleting permissions with a permission ID that does not exist."""
+    invalid_permission_id = permission.id + 256
+    form = DeleteForm(superuser, invalid_permission_id, permission_id=invalid_permission_id,
+                      acknowledged='y')
+
+    assert form.validate() is False
+    assert _('Permission ID "%(permission_id)s" does not exist',
```

Comment:
This is a data leak (since it would enable someone to enumerate the permission IDs), but since we have other checks in place (e.g. cataloging admins can only delete their 'own' permissions) we don't need to worry that much about it.

Just a tip to keep this in mind when writing error messages that might get shown to users.

Reply:
Good point!